File: LeetCode 819 Most Common Word.cpp
// LeetCode 819 Most Common Word.cpp

class Solution {
public:
    string mostCommonWord(string paragraph, vector<string>& banned) {
        unordered_map<string, int> m;
        for (string &str : banned) {
            m[str] = -1;
        }
        
        string buf, ans;
        paragraph += " ";
        for (char ch : paragraph) {
            if (isalpha(ch)) {
                buf += tolower(ch);
            } else if (!buf.empty()) {
                if (m[buf] != -1 && ++m[buf] > m[ans]) {
                    ans = buf;
                }
                buf.clear();
            }
        }
        return ans;
    }
};

File: LeetCode 819 Most Common Word.py
# LeetCode 819 Most Common Word.py
# Environment: Python3

class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        words = re.findall(r'\w+', paragraph.lower())
        permitted = [w for w in words if w not in banned]
        return collections.Counter(permitted).most_common(1)[0][0]